alpine/LandauDampingManager.h

```diff
@@ -116,116 +112,118 @@ class LandauDampingManager
+        fcontainer_m->initializeFields(mesh, FL, solver_m);
 
-        this->fcontainer_m->initializeFields(mesh, FL);
+        setFieldSolver( std::make_shared<FieldSolver_t>(solver_m, &fcontainer_m->getRho(), &fcontainer_m->getE(), &fcontainer_m->getPhi()) );
 
-        this->fsolver_m = std::make_shared<FieldSolver_t>(this->solver, &this->fcontainer_m->getRho(),
-                                                          &this->fcontainer_m->getE());
-        this->fsolver_m->initSolver();
+        fsolver_m->initSolver();
 
-        this->loadbalancer_m = std::make_shared<LoadBalancer_t>(
-            this->lbt, this->fcontainer_m, this->pcontainer_m, this->fsolver_m);
+        setLoadBalancer( std::make_shared<LoadBalancer_t>( lbt_m, fcontainer_m, pcontainer_m, fsolver_m) );
 
-        this->setParticleContainer(pcontainer_m);
+        initializeParticles(mesh, FL);
 
-        this->setFieldContainer(fcontainer_m);
+        static IpplTimings::TimerRef DummySolveTimer  = IpplTimings::getTimer("solveWarmup");
+        IpplTimings::startTimer(DummySolveTimer);
 
-        this->setFieldSolver(fsolver_m);
+        fcontainer_m->getRho() = 0.0;
 
-        this->setLoadBalancer(loadbalancer_m);
+        fsolver_m->runSolver();
 
-        this ->initializeParticles(mesh, FL);
+        IpplTimings::stopTimer(DummySolveTimer);
 
-        this->fcontainer_m->getRho() = 0.0;
+        par2grid();
 
-        this->fsolver_m->runSolver();
+        static IpplTimings::TimerRef SolveTimer = IpplTimings::getTimer("solve");
+        IpplTimings::startTimer(SolveTimer);
 
-        this->par2grid();
+        fsolver_m->runSolver();
 
-        this->fsolver_m->runSolver();
+        IpplTimings::stopTimer(SolveTimer);
 
-        this->grid2par();
+        grid2par();
+
+        dump();
 
         m << "Done";
     }
 
-    void initializeParticles(std::shared_ptr<Mesh_t<Dim>> mesh_m, std::shared_ptr<FieldLayout_t<Dim>> FL_m){
+    void initializeParticles(std::shared_ptr<Mesh_t<Dim>> mesh, std::shared_ptr<FieldLayout_t<Dim>> FL){
         Inform m("Initialize Particles");
 
         using DistR_t = ippl::random::Distribution<double, Dim, 2 * Dim, CustomDistributionFunctions>;
-        double* parR  = new double[2 * Dim];
-        parR[0]       = this->alpha;
-        parR[1]       = this->kw[0];
-        parR[2]       = this->alpha;
-        parR[3]       = this->kw[1];
-        parR[4]       = this->alpha;
-        parR[5]       = this->kw[2];
+        double parR[2 * Dim];
+        for(unsigned int i=0; i<Dim; i++){
+            parR[i * 2   ]  = alpha_m;
+            parR[i * 2 + 1] = kw_m[i];
+        }
         DistR_t distR(parR);
 
-        Vector_t<double, Dim> kw_m     = this->kw;
-        Vector_t<double, Dim> hr_m     = hr;
-        Vector_t<double, Dim> origin_m = origin;
-        if ((this->loadbalancethreshold_m != 1.0) && (ippl::Comm->size() > 1)) {
+        Vector_t<double, Dim> kw     = kw_m;
```

Comment:
I see in so many places you define new temporary variables. Why are they needed? and why can't you directly use here `kw_m`? Note these things increase the lines of code (which we want to minimize) and gives an impression of a complex code even though the code itself might be simple only.

Reply:
Is it because of `KOKKOS_LAMBDA` vs `KOKKOS_CLASS_LAMBDA`? In this case we may then want to not specify so many of them in the class and define some of them just locally rather than having it in the class as well as defining a temporary variable.

Reply:
Yes, I use local variables so that the KOKKOS_LAMBDA doesn't copy all the member variables of the class in the loop. I'll try to reduce local variables as much as possible in the next commit

Reply:
Yes.. also if that is not possible try reducing the number of variables in the class (if they are not used everywhere) and just have them only defined as local variables